hathor/manager.py

```diff
@@ -703,6 +710,35 @@ def generate_parent_txs(self, timestamp: Optional[float]) -> 'ParentTxs':
         can_include = [i.data for i in can_include_intervals]
         return ParentTxs(max_timestamp, can_include, must_include)
 
+    def get_best_blockchain(self, n_blocks: int) -> list['BlockInfo']:
+        """ Get a list with N blocks from the best blockchain
+        in descending order.
+        """
+        if not (n_blocks > 0):
+            raise ValueError(
+                'Invalid N. N must be greater than 0.'
+            )
+
+        block = self.tx_storage.get_best_block()
+        if self._latest_best_blockchain:
+            best_blockchain_tip = self._latest_best_blockchain[0]
+            if block.to_blockinfo() == best_blockchain_tip:
+                return self._latest_best_blockchain
```

Comment:
There's a bug here, if `len(self._latest_best_blockchain)` is not `n_blocks` (because of a previous call with a different value) then it will return a list with the wrong length.

But since you can still make use of the cache when the memoized length is higher, you could do something like:

```python
best_blockchain_tip = self._latest_best_blockchain[0]
if block.hash_hex == best_block and n_blocks <= len(self._latest_best_blockchain):
    return self._latest_best_blockchain[:n_blocks]
```

This also has the benefit of always returning a copy of the cache and thus two different calls wouldn't return the same list object which could accidentally have been used in between. Like this is what would happen if you don't return a copy:

```
x = manager.get_best_blockchain(5)  # returns a list with 5 elements
x.pop()  # the list only has 4 elements now
y = manager.get_best_blockchain(5)  # returns a list with 4 elements
```